### DISCUSSION TITLE
CDN caching causing unauthorized errors?

### DISCUSSION THREAD

**Discussion by jamov** [requarks/wiki#3825] [Help / Questions]
I have the following Wiki.js setup

- Wiki.js (2.4.107) is running inside Docker on an EC2 instance
- CloudFront cached version of Wiki is served from: wiki.example.com
- wiki-origin.example.com serves as the origin for CloudFront

When users access Wiki via wiki.example.com they **sometimes** get unauthorized errors when trying to edit pages. This behaviour seems inconsistent.

Logging out and back in sometimes helps. Other times they have to resort using the uncached version at wiki-origin.example.com - which never causes the authentication errors.

There must be something I'm missing related to user authentication when serving Wiki.js through CloudFront cache (or any other CDN)?

**Comment by jamov** [comment]
Pumping this only once. No one else has had any similar issues?

We're experiencing this with very small userbase and quite rarely, but when opening the wiki to public, and expecting userbase to grow this could pose a significant issue.

**Comment by joeplaa** [comment]
Hey, I know your question is a year old and you might have fixed it in the meantime (or abandoned it altogether). I have a similar setup through CloudFront and haven't had any issues with it. I'm the only user though, so I have a very different usecase.

After some trial and error, I found I had to create a custom cache policy that included headers `Authorization`, `Host`, `CloudFront-Forwarded-Proto` and `X-Forwarded-Proto`. You might need less as I also use this policy for a Ghost blog. I also added cookie `jwt` to the policy. 

Again you might have solved it by now, so in that case this is mostly for other people finding this thread.

**Reply by tigerinus** [reply]
I believe in general, Wiki.js should consider some kind of cache control once after certain changes/contents are done.

For example, including a incrementing counter in cookie/querystring, so that when this counter can be configured as part of the cache key in any CDN, and when it increments, CDN will not return the cached information.

Most CMS platforms have this feature, but it's something missing Wiki.js as far as I know.

**Reply by joeplaa** [reply]
Yes, I have to invalidate the cache every time I make changes, which is really annoying.

**Reply by craigreyenga** [reply]
I was using CloudFront with a default cache policy set, and ran into an issue where logging in would not give me access to the admin area, my profile, nor the ability to log out. Changing the cache settings fixed the problem. Your comment guided me in correct direction. Thanks.
